doc/tutorials/learning_to_rank.rst

```diff
@@ -0,0 +1,177 @@
+
++-------+-----------+---------------+
+|   QID |   Label   |   Features    |
++=======+===========+===============+
+|   1   |   0       |   :math:`x_1` |
++-------+-----------+---------------+
+|   1   |   1       |   :math:`x_2` |
++-------+-----------+---------------+
+|   1   |   0       |   :math:`x_3` |
++-------+-----------+---------------+
+|   2   |   0       |   :math:`x_4` |
++-------+-----------+---------------+
+|   2   |   1       |   :math:`x_5` |
++-------+-----------+---------------+
+|   2   |   1       |   :math:`x_6` |
++-------+-----------+---------------+
+|   2   |   1       |   :math:`x_7` |
++-------+-----------+---------------+
+
+Notice that the samples are sorted based on their query index in an non-decreasing order. Here the first three samples belong to the first query and the next four samples belong to the second. For the sake of simplicity, we will use a pseudo binary learning to rank dataset in the following snippets, with binary labels representing whether the result is relevant or not, and randomly assign the query group index to each sample. For an example that uses a real world dataset, please see :ref:`sphx_glr_python_examples_learning_to_rank.py`.
+
+.. code-block:: python
+
+  from sklearn.datasets import make_classification
+  import numpy as np
+
+  import xgboost as xgb
+
+  # Make a pseudo ranking dataset for demonstration
+  X, y = make_classification(random_state=rng)
+  rng = np.random.default_rng(1994)
+  n_query_groups = 3
+  qid = rng.integers(0, 3, size=X.shape[0])
+
+  # Sort the inputs based on query index
+  sorted_idx = np.argsort(qid)
+  X = X[sorted_idx, :]
+  y = y[sorted_idx]
+
+The simpliest way to train a ranking model is by using the sklearn estimator interface. Continuing the previous snippet, we can train a simple ranking model without tuning:
+
+.. code-block:: python
+
+  ranker = xgb.XGBRanker(tree_method="hist", lambdarank_num_pair_per_sample=8, objective="rank:ndcg", lambdarank_pair_method="topk")
+  ranker.fit(X, y, qid=qid)
+
+Please note that, as of writing, there's no learning to rank interface in sklearn. As a result, the :py:class:`xgboost.XGBRanker` does not fully conform the sklearn estimator guideline and can not be directly used with some of its utility functions. For instances, the ``auc_score`` and ``ndcg_score`` in sklearn don't consider group information nor the pairwise loss. Most of the metrics are implemented as part of XGBoost, but to use sklearn utilities like :py:func:`sklearn.model_selection.cross_validation`, we need to make some adjustments in order to pass the `qid` as an additional parameter for :py:meth:`xgboost.XGBRanker.score`. The `X` for :py:class:`xgboost.XGBRanker` may contain a special column called ``qid`` when it's a pandas dataframe or a cuDF dataframe:
```

Comment:
```suggestion
Please note that, as of writing, there's no learning-to-rank interface in scikit-learn. As a result, the :py:class:`xgboost.XGBRanker` class does not fully conform the scikit-learn estimator guideline and can not be directly used with some of its utility functions. For instances, the ``auc_score`` and ``ndcg_score`` in scikit-learn don't consider query group information nor the pairwise loss. Most of the metrics are implemented as part of XGBoost, but to use scikit-learn utilities like :py:func:`sklearn.model_selection.cross_validation`, we need to make some adjustments in order to pass the ``qid`` as an additional parameter for :py:meth:`xgboost.XGBRanker.score`. Given a data frame ``X`` (either pandas or cuDF), add the column ``qid`` as follows:
```
I revised the last sentence to make it a command, so that it follows clearly from the previous sentence.

Reply:
Good suggestion!